ScratchDjango/User/models.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from django.contrib.auth.models import AbstractUser
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError
+from django.db.models import CharField, EmailField
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from .managers import UserManager
+
+
+def check_email(email):
+    regex = r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b"
+    if re.fullmatch(regex, email):
+        return True
+    raise ValidationError("Enter a valid Email")
+
+
+class User(AbstractUser):
+    # First and last name do not cover name patterns around the globe
+    name = CharField(_("Name of User"), blank=True, max_length=255)
+    first_name = None  # type: ignore
```

Comment:
Why do we need to do it this way?

Reply:
I think a single name field is easy to handle than first name and last name fields separated

Reply:
I agree that in the registeration feature, there's no need of making a custom user model but it was needed in the next features like otp etc.

Reply:
Yes, we can keep these default fields and add extra name field

Reply:
Still not addressed?